spec/controllers/v0/account_controls_controller_spec.rb

```diff
@@ -0,0 +1,288 @@
+  end
+
+  describe 'GET csp_index' do
+    let(:icn) { user_account.icn }
+    let(:icn_param) { icn }
+
+    context 'without an ICN param' do
+      let(:icn_param) { nil }
+      let(:expected_error) { 'The required parameter "icn", is missing' }
+
+      it 'returns a bad request error' do
+        get :csp_index, params: { icn: icn_param }
+
+        expect(response).to have_http_status(:bad_request)
+        error = JSON.parse(response.body)['errors'].first
+        expect(error['detail']).to eq(expected_error)
+      end
+    end
+
+    context 'with an ICN param' do
+      context 'when a record is not found' do
+        let(:icn_param) { 'some-bad-icn' }
+        let(:expected_error) { "User CSP Index not found. ICN:#{icn_param}" }
+
+        it 'returns a not found error' do
+          get :csp_index, params: { icn: icn_param }
+
+          expect(response).to have_http_status(:not_found)
+          error = JSON.parse(response.body)['error']
+          expect(error).to eq(expected_error)
+        end
+      end
+
+      context 'when one or more User Verifications are found' do
+        let(:expected_csp_uuids) do
+          [logingov_user_verification.logingov_uuid,
+           idme_user_verification.idme_uuid,
+           dslogon_user_verification.dslogon_uuid,
+           mhv_user_verification.mhv_uuid]
+        end
+
+        before do
+          logingov_user_verification
+          idme_user_verification
+          dslogon_user_verification
+          mhv_user_verification
+        end
+
+        it 'returns serialized user account data' do
+          get :csp_index, params: { icn: icn_param }
+
+          expect(response).to have_http_status(:ok)
+          serialized_data = JSON.parse(response.body)['data']
+          expect(serialized_data['icn']).to eq(icn)
+          expect(serialized_data['csp_verifications'].count).to eq(4)
+          serialized_data['csp_verifications'].each do |csp_verification|
+            expect(csp_verification['type']).to be_in(%w[logingov idme dslogon mhv])
+            expect(csp_verification['csp_uuid']).to be_in(expected_csp_uuids)
+          end
+        end
+      end
+    end
+  end
+
+  shared_context 'when validating params and querying a UserVerification' do
+    shared_examples 'when a record is not found' do
+      let(:expected_error_message) { "User CSP record not found. #{type_param}_uuid:#{csp_uuid_param}" }
+
+      it 'returns a not found error' do
+        post lock_action, params: account_controls_params
+        expect(JSON.parse(response.body)['error']).to eq(expected_error_message)
+      end
+    end
+
+    shared_examples 'when a record is found' do
+      context 'when the record is successfully updated' do
+        it 'performs the requested lock action' do
+          post lock_action, params: account_controls_params
+
+          user_verification = UserVerification.find_by("#{type}_uuid": csp_uuid)
```

Comment:
Few things to make this spec more straightforward:
- define `subject { post lock_action, params: account_controls_params }`
- put `user_verification` definition here in a `let` instead of inside the `it` block
- This will let you do `expect { subject }.to change(user_verification.locked).from(false).to(true)` or something like that